### DISCUSSION TITLE
usePanStrategy issue

### DISCUSSION THREAD

**Discussion by FTInterritus7624** [samuelarbibe/dnd-timeline#28] [Q&A] [answered]
Hello, I was trying out usePanStrategy within the TimelineContext and I noticed that whenever I used useDragStrategy or useWheelStrategy, my console would give:

"[Violation] Added non-passive event listener to a scroll-blocking <some> event. Consider marking event handler as 'passive' to make the page more responsive"

Tracing it to its file, it would point to violation being caused by **element.addEventListener("wheel", pointerWheelHandler);** within both useDragStrategy and useWheelStrategy. 

1. Is there a way to resolve this?
2. As an alternative to using the built in usePanStrategies, does dnd-timeline offer/allow a way just pan horizontally with a horizontal scrollbar? Or maybe a simple click and drag on the timeline (granted I understand, I might have to disable draggable for the boxes) without using keyboard input?

**Comment by samuelarbibe** [comment]
Hey!
I also got this error, which is on the verbose level.  
It calls recommends using an option called `passive` on the scroll event, to explicitly define if the event's default behaviour is going to be prevented.  This is to warn the developer from mistakenly preventing the default scrolling behaviour.  
  
In this case, the default scroll behaviour is purposly overriden and replaced by a custom mechanism, so all there is to do is to explicity define that default behaviour is going to be overriden to suppress this verbose log.  

Reference:
https://github.com/WICG/EventListenerOptions/blob/gh-pages/explainer.md  
 
This issue is going to be fixed as a patch in the next version. For now, it can be safetly ignored.

**Reply by samuelarbibe** [reply]
Resolved in https://github.com/samuelarbibe/dnd-timeline/pull/30

**Comment by samuelarbibe** [comment, marked as answer]
As for building a custom pan strategy: You can build any strategy you want. 
please reference this documentation:  
https://samuel-arbibe.gitbook.io/dnd-timeline/documentation/zoom-and-pan

But, keep in mind, the timeframe is coupled to a react state. Timeframe defines the horizontal viewport of timeline, and it cannot be controlled on the CSS level using the default scrolling mechanism.  
Therfore, you will need to create a custom strategy by listening to any event of your choise, and calling the `onPanEnd` callback when you want.  

Best of luck 🤩
